plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -1880,11 +1900,10 @@ private void finishWrite(ConnectorSession session, IcebergTableHandle table, Col
         if (!fullyDeletedFiles.isEmpty()) {
             try {
                 TrinoFileSystem fileSystem = fileSystemFactory.create(session);
-                for (List<CommitTaskData> commitTasksToCleanUp : fullyDeletedFiles.values()) {
-                    for (CommitTaskData commitTaskData : commitTasksToCleanUp) {
-                        fileSystem.deleteFile(commitTaskData.getPath());
-                    }
-                }
+                fileSystem.deleteFiles(fullyDeletedFiles.values().stream()
```

Comment:
You may want to make the multi-file deletion in batches of `DELETE_BATCH_SIZE` size.

Reply:
+1

Reply:
I agree that the way you did it here is not helpful. Actually I expected you will partition it on higher level - 
```
fullyDeletedFiles.values().stream()
                        .flatMap(Collection::stream)
                        .map(CommitTaskData::getPath)
``` 
without collecting it to list first. But now, after I learnt that `Lists.partition()` is not lazy, I am not sure if this can be done...